lib/postgrex/protocol.ex

```diff
@@ -2653,8 +2653,19 @@ defmodule Postgrex.Protocol do
   defp decode_tag(<<h, t::binary>>, acc) when h in ?A..?Z,
     do: decode_tag(t, <<acc::binary, h + 32>>)
 
-  defp decode_tag(<<h, t::binary>>, acc),
-    do: decode_tag(t, <<acc::binary, h>>)
+  # Valid SQL statements in Postgresql are only
+  # uppercase A..Z and space. Therefore any other
+  # character prompts a return of the accumulator
+  # ignoring anything from the invalid character
+  # and any trailing space.
+  defp decode_tag(<<_h, _t::binary>>, acc) do
+    tag =
+      acc
+      |> String.trim_trailing(" ")
```

Comment:
What if we have a clause that is explicit about spaces then? Or just explicitly match on something like `decode_tag(<<h, ?\s, t::binary>>, acc) when in ?A..?Z` and remove the accumulator building altogether? :)

Reply:
José there are several SQL commands that have `\s` in them of course so the `\s` isn't terminal - its a character outside of `A..Z` and `\s` that is.  By the time we discover a terminal character there may already be a trailing space (actually trailing `_` since it will already have been transliterated). Or I am I just being slow at 3:49am 😄